### PR TITLE
front: make rs label shorter

### LINKED ISSUE #9358: stdcm: make rolling stock labels shorter
### Who would benefit from this feature?

End users

### What is this feature about?

In the list of rolling stocks here, the labels are very long, we can make them shorter by only displaying the reference of the rolling stock:

Image

From the payload the front gets, display the "name" and "series" if it exists, "reference" otherwise.
```
{
        "name": "JT42US",
        "detail": "CLASS 66",
        "reference": "JT42",
        "series": "CLASS 66"
}
```

### Why is this feature valuable?

This feature matters because it will make the rolling stock list clearer.

### PR DESCRIPTION
**Pull request by kmer2016**
close #9358

### CODE REVIEW

**Review by axrolld — APPROVED** [review]
The logic's right and i saw what it looked like, it's great ! Thanks 😁

File: front/src/applications/stdcm/components/StdcmForm/StdcmConsist.tsx [hunk 1]
```diff
@@ -72,22 +72,8 @@ const StdcmConsist = ({ disabled = false }: StdcmConfigCardProps) => {
...
-          ? metadata.subseries
-          : metadata.detail;
-      if (series) res += series;
-      if (subseries) res += series ? ` (${subseries})` : subseries;
-    }
-    if (name) res += ` -- ${name}`;
-    return res;
-  };
+  const getLabel = (rs: LightRollingStockWithLiveries) =>
+    `${rs.name} - ${rs.metadata?.series || rs.metadata?.reference || ''}`;
```
Comment (clarani) [inline front/src/applications/stdcm/components/StdcmForm/StdcmConsist.tsx]: If we don't have any metadata, we will have the '-', is it on purpose ?

**Review by Akctarus — APPROVED** [review]
Tested, LGTM

File: front/src/applications/stdcm/components/StdcmForm/StdcmConsist.tsx (hunk 1 above)
Comment (kmer2016) [reply front/src/applications/stdcm/components/StdcmForm/StdcmConsist.tsx]: Nope, I pushed the fix

**Review by clarani — APPROVED** [review]
LGTM :white_check_mark: Can you squash your commits ?

**Pull request merged**